src/plugins/explore/public/application/hooks/editor_hooks/use_language_switch/use_language_switch.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { EditorMode } from '../../../utils/state_management/types';
+import { setEditorMode } from '../../../utils/state_management/slices';
+import { useEditorRef } from '../use_editor_ref';
+
+export interface UseLanguageSwitchReturn {
```

Comment:
nitpick: instead of `useLanguageSwitch`, `useSwitchLanguage`? `useLanguageSwitch` sounds like yoda